subprojects/core/src/main/java/org/gradle/process/internal/worker/WorkerProcessBuilder.java

```diff
@@ -64,6 +64,16 @@ public interface WorkerProcessBuilder extends WorkerProcessSettings {
 
     NativeServicesMode getNativeServicesMode();
 
+    /**
+     * Set if Gradle should add JPMS-compatibility flags to the worker process when needed.
+     * These are needed on Java 24+ when using native services because they access native libraries.
+     * There may be other cases in the future where different flags are needed in other scenarios.
+     *
+     * @param addJpmsCompatibilityFlags whether to add the flags
+     * @return this
+     */
+    WorkerProcessBuilder setAddJpmsCompatibilityFlags(boolean addJpmsCompatibilityFlags);
```

Comment:
Is this being called from anywhere?

Reply:
No, I added it in case we needed to call it from somewhere (e.g. testing used to set a similar flag for `--add-opens`).